src/saveload/afterload.cpp

```diff
@@ -929,19 +929,18 @@ bool AfterLoadGame()
 						break;
 
 					case STATION_OILRIG: {
+						/* The internal encoding of oil rigs was changed twice.
+						 * It was 3 (till 2.2) and later 5 (till 5.1).
+						 * Setting it unconditionally does not hurt.
+						 */
+						Station::GetByTile(t)->airport.type = AT_OILRIG;
```

Comment:
Can you modify the last sentence of the comment to something like this which explains why it's moved out of the if/else block like before:
DeleteOilRig asserts on the correct type, and setting it unconditionally does not hurt.